src/toolbox.ts

```diff
@@ -564,14 +562,121 @@ export class RecolorActiveItem extends ToolboxItem {
                     <input type="button" class="color-change-btn" id="color-change-cya">
                 </div>
                 <div class="color-picker-container" id="color-picker-container">
-                    <input type="color"  class="color-change-picker" id="color-change-pick">
+                    <input type="color" class="color-change-picker" id="color-change-pick">
                 </div>
             </div>
         </div>
         `
     }
 }
 
+export class KeypointSlider extends ToolboxItem {
+    public html: string;
+    public inner_HTML: string;
+
+    constructor(ulabel: ULabel, get_annotation_confidence: Function) {
+        super();
+        this.inner_HTML = `<p class="tb-header">Keypoint Slider</p>`;
+        $(document).on("input", "#keypoint-slider", (e) => {
+            var current_subtask_key = ulabel.state["current_subtask"];
+            var current_subtask = ulabel.subtasks[current_subtask_key];
+            $("#keypoint-slider-label").text(e.currentTarget.value + "%")
+
+            const annotation_confidence = get_annotation_confidence(current_subtask)
+            
+            //this.draw_histogram(current_subtask, annotation_confidence)
+            this.update_annotations(current_subtask, annotation_confidence , e.currentTarget.value / 100)
+            ulabel.redraw_all_annotations(null, null, false);
+        })
+    }
+    
+    //annotation_confidence should be in the form [{id: "", confidence: "", class_id: ""}, {id: "", confidence: "", class_id: ""}, ...]
+    public update_annotations(subtask, annotation_confidence, filter_value) {
+
+        for (let annotation in annotation_confidence) {
+            if (annotation_confidence[annotation].confidence < filter_value) {
+                subtask.annotations.access[annotation_confidence[annotation].id].deprecated = true
+            }
+            if (annotation_confidence[annotation].confidence >= filter_value) {
+                subtask.annotations.access[annotation_confidence[annotation].id].deprecated = false
+            }
```

Comment:
Have this be a function that operates on a single annotation that gets passed into this toolboxitem.
Put it in the same file as the other function